Add campaign routes

CCBet opened the pull request.

CCBet committed:
Added the Id property for the asset object and wrote the DeleteAssetTest

CCBet committed:
Removed unused variable

CCBet committed:
Worked on adding campaign routes

CCBet committed:
Refactored the DeleteAssetTest

CCBet committed:
Replaced Asset class name with {{classname}} in the api.mustache file

CCBet committed:
Changed a mustache file, added campaign routes

This branch us created from the delete-asset-test branch. On that branch I modified the api.mustache file and that change had a bug so I changed this file again on this branch in order to revert it to the initial state. I added routes for working with Campaign objects (for creating/returning a Campaign object)

CCBet committed:
This commit reverts a change made on the api.mustache file in a previous commit

This commit reverts a change made on the api.mustache file in a previous commit because that change was not helpful.

Commit:
Merge branch 'add-campaign-routes' of https://github.com/CCBet/swagger-codegen-playground into add-campaign-routes

Commit:
Merge branch 'delete-asset-test' into add-campaign-routes

Commit:
Deleted two .cs files and renamed a test function.

Deleted two .cs files that were the result of code generation over a typo in the OpenAPI spec. Changed the name of the DeleteAssetTest to DeleteAssetByIdTest.

sfdrogojan commented on openapi-2.0/generated/src/IO.Swagger.Test/Api/AssetApiTests.cs:
Should be split in 2 tests:

1. Create Asset, Delete it, Retrieve it - assert on the retrieval of the deleted Asset.
2. Multiple deletes of the same Asset.

sfdrogojan commented on openapi-2.0/generated/src/IO.Swagger.Test/Api/CampaignApiTests.cs:
```diff
@@ -0,0 +1,103 @@
+
+            Assert.AreEqual(name, response.Name);
+            Assert.AreEqual(description, response.Description);
+            Assert.AreEqual(campaignCode.Replace(" ",""), response.CampaignCode); // the server removes all spaces from the campaign code
```
Use the expected string without logic.

sfdrogojan commented on openapi-2.0/generated/src/IO.Swagger.Test/Api/CampaignApiTests.cs:
```diff
@@ -0,0 +1,103 @@
+        [Test]
+        public void GetCampaignByIdTest()
+        {
+            var name = $"Campaign name";
```
Create factory method to get a Campaign.

Commit:
Implemented the required changes

Created factories for Campaign and Asset objects, removed the logic for eliminating white spaces in the CampaignApiTests.cs, created the DeleteNonExistingAssetTest

Commit:
Refactored three tests in AssetApiTests.cs

Refactored three tests from AssetApiTests.cs so that the posted object will be deleted if one of the asserts failed.

sfdrogojan reviewed (approved):
Code review done in a pair programming session.
